Molly chat improvements

MegumiKim opened the pull request:
Molly frontend tasks I am working on can be divided to several stages.
I would like to request reviews at each sub tasks I am working on.

First commits encompass the following changes:
 - Add Label to textarea
 - Prevent empty query to be sent
 - Enable press enter query submit
 - Unhide textarea for mobile view

One question: 
Now that query can be submitted on press Enter key, can I remove submit button in order to secure more space for textarea and avoid confusion?

StianSto commented:
great work as always :) tested it on my phone, and while molly wasnt visible at first (fixable by adjusting height on the wrapper) i was able to write and send a query, and got a response. im also not able to send an empty query. 

requests: 
- [ ] could you add the new changes to +server.ts in molly package?

MegumiKim commented:
@StianSto 
Thanks Stian for testing.
As requested, uncommented +server.ts file.

StianSto commented on packages/molly/src/routes/+server.ts:
```diff
@@ -1,6 +1,6 @@
 
-// export const GET = molly.handlers.chatCompletion.GET;
\ No newline at end of file
+export const GET = molly.handlers.chatCompletion.GET;
```
```suggestion
export const POST = molly.handlers.chatCompletion.POST;
```

MegumiKim replied:
Sorry for my oversight! Thanks for the suggestion :-)

StianSto reviewed (changes requested):
new backend uses POST. other than that i can't see a reason this shouldn't be good to go :)

MegumiKim commented:
Molly chat box should now automatically scrolls to the bottom.

MegumiKim commented:
- Molly's chat box is now automatically scrolled to the bottom.
- I have removed the submit button to see if it is better. Please let me know if you encounter any issue.

Fermain commented on packages/molly/src/lib/components/MollyForm.svelte:
```diff
@@ -1,24 +1,42 @@
+			placeholder="Your Query"
+			id="userText"
+			on:keypress={(event) => handleKeyPress(event)} />
+			<!-- <button type="submit" class="btn variant-filled-primary rounded-none">
```
Can you remove the dead code?

Fermain commented on packages/molly/src/lib/components/MollyHeader.svelte:
```diff
@@ -4,11 +4,8 @@
 	let img: string = closeButton;
 
-	export let isOpen = false;
+export let toggleMollyOpen:()=>void;
```
It would be awesome if this focused the text box

Fermain commented on packages/molly/src/lib/components/MollyMessages.svelte:
```diff
@@ -27,4 +29,5 @@
 	{/if}
-</div>
\ No newline at end of file
+</div>
```
Formatting/indentation?

MegumiKim replied:
Sorry I forgot to run format.

Fermain reviewed (changes requested):
Excellent work! Just a few minor changes and one new request

MegumiKim commented:
Please note the following changes:

- Re-organized Molly components (now Molly chat box does not live inside Molly button).
- Added custom resize to Molly chat box.
- Replace the close button to Molly icon to simplify the design. Since same icon opens the chat box, I believe this is intuitive to user...
- Added hover state (drop shadow) to Molly icon buttons. Can be less subtle if it is required. 
- Added aria-label to Molly icon btns

Anclagen commented:
There are some issues when it comes to resizing the box.

On Firefox Molly is expanding with the text.
![molly-firefox](https://github.com/Fermain/-mollify/assets/91462616/1785ff69-b10d-4e99-894e-247dccfd9ca1)

On chrome its fine till you resize it.
![molly-chrrome](https://github.com/Fermain/-mollify/assets/91462616/70d59f2c-a63e-449c-806a-62a26ef15339)

Fermain commented:
We should bind escape to closing Molly. when fully expanded, it sits behind the header meaning you can't close it.

Would be useful to have a small explainer like:

> Enter to send. Shift + Enter for new line.

MegumiKim commented:
@Anclagen Thanks for spotting. 
There was a bug in the last commit. Should be ok now...

MegumiKim commented:
In addition to custom resize, I added fixed resize button with placeholder icon. 
Please test if it works ok/good size.

pretzL commented on packages/molly/src/lib/components/Molly.svelte:
I wanted to add this as a suggestion because it might be a little hard to follow the explanation, but Git didn't want to let me do that easily so I'll try to explain. Feel free to reach out if itæs unclear.

Instead of getting the container element like so
```js
const container = document.querySelector<HTMLElement>('.chat-container')
```
You can use Svelte's own `bind` directive to reference the element, like so
```svelte
<div bind:this={container} class="chat-container etc...
```

You'd then define the `container` variable simply in the top of the script tag like so:
```svelte
let container:<HTMLElement> = null;
```

Fermain replied:
This guy sveltes

pretzL replied:
😎

pretzL commented on packages/molly/src/lib/components/MollyButton.svelte:
```diff
@@ -1,22 +1,13 @@
-<script>
-	import MollyHeader from './MollyHeader.svelte';
+<script lang='ts'>
+	// import MollyHeader from './MollyHeader.svelte';
```
Don't leave commented out code in the PR.
```suggestion
```

pretzL reviewed (changes requested):
There are some formatting issues throughout the PR, which should be resolved before merging. See also my comments.

MegumiKim commented:
@StianSto @pretzL @Fermain 
All the requested changes are now made. Hope you could approve :-)

Fermain reviewed (approved):
I did some fiddling around with formatting rules, but it's now fixed and in line with the LMS formatting rules.
